tools/diagnostics/nsdiff

```diff
@@ -0,0 +1,673 @@
+        return tempdir
+    
+    # Extract a tarball/zipfile into a temprary directory
+    def _extract_archive(self, file):
+        logger.debug("Extracting archive {}".format(file))
+        # Is the file a tarfile (tar, gz2, zip, tar.gz, etc.) or zipfile?
+        if tarfile.is_tarfile(file):
+            return self._extract_tarball(file)
+        elif zipfile.is_zipfile(file):
+            return self._extract_zipfile(file)
+        else:
+            raise ArchiveError("{} must be a tarfile or a zipfile".format(file))
+    
+    def split_name(self, file):
+        abs_file = abspath(file)
+        return (
+            dirname(abs_file),
+            basename(abs_file),
+            splitext(basename(abs_file))[0],
+            splitext(basename(abs_file))[1]
+        ) 
+    
+    def transform_and_scrub(self, sin, sout):
+        """Convert all state data to a *nix diff-able form
+    
+        Keyword arguments:
+        sin -- A directory containing indy-node/plenum state data
+        sout -- A directory in which to write *nix-diff-able/friendly output
+    
+        Detect all LevelDB/RocksDB directories. The LevelDB directory structure
+        is described here:
+            https://github.com/google/leveldb/blob/master/doc/impl.md 
+
+            Each database is represented by a set of files stored in a
+            directory.
+       
+            By default, LevelDB uses a BytewiseComparator, which is the expected
+            outcome when calling initKeyValueStorage to create a LevelDB
+            instance/directory. initKeyValueStorageIntKeys changes the
+            comparator to an IntegerComparator. Therefore, the LevelDB
+            instances/directories may require one or the other. First Attempt to
+            open with initKeyValueStorage. If that throws the following error,
+            try again with initKeyValueStorageIntKeys.
+                "leveldb.LevelDBError: Invalid argument: IntegerComparator does
+                 not match existing comparator : leveldb.BytewiseComparator"
+       
+        Transform and scrub data in preparation for *nix diff
+           1. Marshall/Transform data from LevelDB directories to flat files.
+              Use domain objects where possible to decode key/value pairs.
+           2. Any subset of 'particpants' (nodes) can be used to create
+              state 'signature'(s). Therefore, scrub state_signature LevelDB
+              entries and exclude 'participants' and 'signature' fields;
+              preserving only the 'value' JSON element.
+           3. Scrub the node_info; preserving only the 'view' field.
+           4. Extract the last/latest (view no, ppSeqNo) tuple from the logs
+              emitted by replica.py and node.py. TODO: update this description
+              once we have concrete examples.
+           5. All other files and directories are unaltered.
+        """
+        logger.debug("Transforming and scrubbing {} into {}".format(sin, sout))
```

Comment:
We migrated to RocksDB. RocksDB is now the default DB for all states/ledgers/storages.